lxd/instance.go

```diff
@@ -633,3 +633,35 @@ func pruneExpiredAndAutoCreateInstanceSnapshotsTask(d *Daemon) (task.Func, task.
 
 	return f, schedule
 }
+
+func getSourceImageFromInstanceSource(ctx context.Context, s *state.State, tx *db.ClusterTx, project string, source api.InstanceSource, sourceImageRef *string, instType string) (*api.Image, error) {
```

Comment:
Is `sourceImageRef` equivalent to `imageName` in the other functions (in that it can take a fingerprint or an alias), if so then I quite like it and think we should use `imageRef` in the other 2 functions rather than `imageName`.